### PR TITLE
Adding functions examples for azure-nextgen

### PR DESCRIPTION
**Pull request by viveklak**

### CODE REVIEW

File: azure-nextgen-ts-functions-raw/index.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    name: "httpdotnet",
+    resourceGroupName: resourceGroup.name,
+    serverFarmId: plan.id,
+    kind: "FunctionApp",
+    siteConfig: {
+        appSettings: [
+            { name: "runtime", value: "dotnet" },
+            { name: "FUNCTIONS_WORKER_RUNTIME", value: "dotnet" },
+            { name: "WEBSITE_RUN_FROM_PACKAGE", value: dotnetBlobSignedURL },
+            { name: "WEBSITE_NODE_DEFAULT_VERSION", value: "~12" },
```
Comment (mikhailshilkov) [inline azure-nextgen-ts-functions-raw/index.ts:60]: Probably not needed for non-node functions?

File: azure-nextgen-ts-functions-raw/dotnet/functionapp-dotnet.csproj [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <AzureFunctionsVersion>v2</AzureFunctionsVersion>
```
Comment (mikhailshilkov) [inline azure-nextgen-ts-functions-raw/dotnet/functionapp-dotnet.csproj:4]:
```suggestion
    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
```

**Review by mikhailshilkov — APPROVED** [review]
LGTM, just minor suggestions

File: azure-nextgen-cs-functions-linux-app/FunctionsStack.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
+// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.
```
Comment (ericrudder) [inline azure-nextgen-cs-functions-linux-app/FunctionsStack.cs:1]: 2021, here and elsewhere?

File: azure-nextgen-ts-functions-raw/python/HelloPython/__init__.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+def main(req: func.HttpRequest) -> func.HttpResponse:
+    logging.info('Python HTTP trigger function processed a request.')
+
+    name = req.params.get('name')
+    if not name:
+        try:
+            req_body = req.get_json()
+        except ValueError:
+            pass
+        else:
```
Comment (lblackstone) [inline azure-nextgen-ts-functions-raw/python/HelloPython/__init__.py:15]: I think this is indented too far

**Review by lblackstone — APPROVED** [review]
Looks good aside from one comment

File: azure-nextgen-cs-functions-linux-app/FunctionsStack.cs [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+        var app = new WebApp("app", new WebAppArgs
+        {
+            Name = "function-app-py",
+            ResourceGroupName = resourceGroup.Name,
+            ServerFarmId = appServicePlan.Id,
+            SiteConfig = new SiteConfigArgs
+            {
+                AlwaysOn = true,
+                LinuxFxVersion = "DOCKER|mcr.microsoft.com/azure-functions/python:2.0",
```
[the remark this answers is not in the document]
Comment (viveklak) [reply azure-nextgen-cs-functions-linux-app/FunctionsStack.cs:85]: This was copied from https://github.com/pulumi/examples/blob/master/azure-cs-functions-linux-app/FunctionsStack.cs#L82
Seems to be the base functions runtime image and it seemed to run our code still. However, doesn't work with the addition of the consumption sku. Removed now.

File: azure-nextgen-ts-functions-raw/python/HelloPython/__init__.py (hunk 4 above)
Comment (viveklak) [reply azure-nextgen-ts-functions-raw/python/HelloPython/__init__.py:15]: This is a shameless copy of https://github.com/pulumi/examples/blob/master/azure-ts-functions-raw/python/HelloPython/__init__.py. 

That said I think the intent is that the `else` block is invoked only if no exception is raised (as opposed to a finally block).

File: azure-nextgen-cs-functions-linux-app/FunctionsStack.cs (hunk 3 above)
Comment (viveklak) [reply azure-nextgen-cs-functions-linux-app/FunctionsStack.cs]: Thanks for the catch, updated.

File: azure-nextgen-ts-functions-raw/dotnet/functionapp-dotnet.csproj (hunk 2 above)
Comment (viveklak) [reply azure-nextgen-ts-functions-raw/dotnet/functionapp-dotnet.csproj:4]: Fixed.

File: azure-nextgen-ts-functions-raw/index.ts (hunk 1 above)
Comment (viveklak) [reply azure-nextgen-ts-functions-raw/index.ts:60]: Yup - removed.

File: azure-nextgen-cs-functions-linux-app/FunctionsStack.cs [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+                },
+            },
+        });
+
+        this.Endpoint = Output.Format($"https://{app.DefaultHostName}/api/Hello?name=Pulumi");
+    }
+
+    [Output] public Output<string> Endpoint { get; set; }
+
+    static private Output<string> signedBlobReadUrl(Blob blob, BlobContainer container, StorageAccount account, ResourceGroup resourceGroup)
```
[the remark this answers is not in the document]
Comment (viveklak) [reply azure-nextgen-cs-functions-linux-app/FunctionsStack.cs]: Fixed.

**Pull request merged**